File: java/src/test/java/com/example/livecoding/strings/ValidAnagramTest.java
package com.example.livecoding.strings;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

class ValidAnagramTest {

    @Test
    void trueCase() {
        assertTrue(ValidAnagram.validAnagram("anagram", "nagaram"));
    }

    @Test
    void falseDifferentLengths() {
        assertFalse(ValidAnagram.validAnagram("rat", "car"));
    }

    @Test
    void empty() {
        assertTrue(ValidAnagram.validAnagram("", ""));
    }

    @Test
    void caseMatters() {
        assertFalse(ValidAnagram.validAnagram("a", "A"));
    }
}

File: typescript/tests/validAnagram.test.ts
import { describe, expect, it } from "vitest";

import { validAnagram } from "../src/strings/validAnagram.js";

describe("validAnagram", () => {
  it("true", () => {
    expect(validAnagram("anagram", "nagaram")).toBe(true);
  });

  it("false", () => {
    expect(validAnagram("rat", "car")).toBe(false);
  });

  it("empty", () => {
    expect(validAnagram("", "")).toBe(true);
  });

  it("case sensitive", () => {
    expect(validAnagram("a", "A")).toBe(false);
  });
});
